Kokkos math namespace error for Kokkos>=4.0

Hi,

I am trying to install n-body solver. I have followed these steps,

```
adepudinesh@UoSDellLaptop (master) /home/adepudinesh/Kokkos/HACCabana/build $  
|  DELL LAPTOP UoS => cmake -D CMAKE_PREFIX_PATH=$CABANA_INSTALL_DIR -D CMAKE_PREFIX_PATH=$KOKKOS_INSTALL_DIR -D CMAKE_PREFIX_PATH=/home/adepudinesh/Kokkos/Silo .. 
-- Enabled Kokkos devices: OPENMP;SERIAL
-- Found SILO: /home/adepudinesh/Kokkos/Silo/include  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/adepudinesh/Kokkos/HACCabana/build
 
```
This successfully works without any errors. However, while I run `make`, I get the following errors,

```
adepudinesh@UoSDellLaptop (master) /home/adepudinesh/Kokkos/HACCabana/build $  
|  DELL LAPTOP UoS => make
[ 12%] Building CXX object src/CMakeFiles/HACCabana.dir/Parameters.cxx.o
[ 25%] Building CXX object src/CMakeFiles/HACCabana.dir/ParticleActions.cxx.o
/home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.cxx: In lambda function:
/home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.cxx:118:61: error: ‘sqrt’ is not a member of ‘Kokkos::Experimental’
  118 |               const float tmp =  1.0f/Kokkos::Experimental::sqrt(dist2Err*dist2Err*dist2Err) - FGridEvalPoly(dist2);
      |                                                             ^~~~
/home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.cxx:118:61: note: suggested alternatives:
In file included from /usr/include/features.h:486,
                 from /usr/include/x86_64-linux-gnu/c++/11/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/11/bits/c++config.h:586,
                 from /usr/include/c++/11/string:38,
                 from /home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.h:4,
                 from /home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.cxx:1:
/usr/include/x86_64-linux-gnu/bits/mathcalls.h:143:1: note:   ‘sqrt’
  143 | __MATHCALL (sqrt,, (_Mdouble_ __x));
      | ^~~~~~~~~~
In file included from /home/adepudinesh/Kokkos/kokkos/build/install/include/Kokkos_Complex.hpp:28,
                 from /home/adepudinesh/Kokkos/kokkos/build/install/include/Kokkos_MathematicalSpecialFunctions.hpp:31,
                 from /home/adepudinesh/Kokkos/kokkos/build/install/include/Kokkos_Core.hpp:54,
                 from /home/adepudinesh/Kokkos/Cabana/build/install/include/Cabana_Types.hpp:19,
                 from /home/adepudinesh/Kokkos/Cabana/build/install/include/Cabana_Slice.hpp:19,
                 from /home/adepudinesh/Kokkos/Cabana/build/install/include/Cabana_AoSoA.hpp:20,
                 from /home/adepudinesh/Kokkos/Cabana/build/install/include/Cabana_Core.hpp:21,
                 from /home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.h:6,
                 from /home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.cxx:1:
/usr/include/c++/11/complex:929:5: note:   ‘std::sqrt’
  929 |     sqrt(const complex<_Tp>& __z) { return __complex_sqrt(__z.__rep()); }
      |     ^~~~
In file included from /home/adepudinesh/Kokkos/kokkos/build/install/include/Kokkos_MathematicalSpecialFunctions.hpp:31,
                 from /home/adepudinesh/Kokkos/kokkos/build/install/include/Kokkos_Core.hpp:54,
                 from /home/adepudinesh/Kokkos/Cabana/build/install/include/Cabana_Types.hpp:19,
                 from /home/adepudinesh/Kokkos/Cabana/build/install/include/Cabana_Slice.hpp:19,
                 from /home/adepudinesh/Kokkos/Cabana/build/install/include/Cabana_AoSoA.hpp:20,
                 from /home/adepudinesh/Kokkos/Cabana/build/install/include/Cabana_Core.hpp:21,
                 from /home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.h:6,
                 from /home/adepudinesh/Kokkos/HACCabana/src/ParticleActions.cxx:1:
/home/adepudinesh/Kokkos/kokkos/build/install/include/Kokkos_Complex.hpp:726:50: note:   ‘Kokkos::sqrt’
  726 | KOKKOS_INLINE_FUNCTION Kokkos::complex<RealType> sqrt(
      |                                                  ^~~~
make[2]: *** [src/CMakeFiles/HACCabana.dir/build.make:90: src/CMakeFiles/HACCabana.dir/ParticleActions.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: src/CMakeFiles/HACCabana.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
 
```

I am not sure why this is failing, I was able to compile and build `ExaMPM` with current `Kokkos` and `Cabana` installation. How do I fix this?

Many thanks.

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -23,7 +23,7 @@ jobs:
         uses: actions/checkout@v2.2.0
         with:
           repository: kokkos/kokkos
-          ref: 3.6.01
+          ref: 3.7.01
           path: kokkos
       - name: Build kokkos
         working-directory: kokkos

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,9 @@ project(HACCabana LANGUAGES CXX C)
 
 include(GNUInstallDirs)
 
-find_package(Cabana)
+find_package(Cabana REQUIRED)
+if(Kokkos_VERSION_MAJOR LESS 4 AND Kokkos_VERSION_MINOR LESS 7)
+  message( FATAL_ERROR "HACCabana requires Kokkos >= 3.7" )
+endif()
 
 add_subdirectory(src)

--- a/src/ParticleActions.cxx
+++ b/src/ParticleActions.cxx
@@ -115,7 +115,7 @@ void ParticleActions::updateVel(\
             if (dist2 < rmax2) 
             {
               const float dist2Err = dist2 + rsm2;
-              const float tmp =  1.0f/Kokkos::Experimental::sqrt(dist2Err*dist2Err*dist2Err) - FGridEvalPoly(dist2);
+              const float tmp =  1.0f/Kokkos::sqrt(dist2Err*dist2Err*dist2Err) - FGridEvalPoly(dist2);
               force[0] += dx * tmp;
               force[1] += dy * tmp;
               force[2] += dz * tmp;

